game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/CombatValue.java

```diff
@@ -0,0 +1,98 @@
+package games.strategy.triplea.delegate.power.calculator;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.TerritoryEffect;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.delegate.Matches;
+import java.util.Collection;
+
+public interface CombatValue {
+
+  StrengthAndRollCalculator getRoll();
+
+  StrengthAndRollCalculator getStrength();
+
+  boolean isDefending();
+
+  GameData getGameData();
+
+  static CombatValue buildMain(
+      final Collection<Unit> allEnemyUnitsAliveOrWaitingToDie,
+      final Collection<Unit> allFriendlyUnitsAliveOrWaitingToDie,
```

Comment:
side-note, 'waiting to die' is not very meaningful. We should consider renaming that to 'selectedAsCasualties' or simply 'casualties'. In the game, a 'casualty' is relatively well defined to be the units selected as casualties and able to fire back. The removal of casualties is a distinct phase, offense of course removes casualties immediately. 'waiting to die' I think is adding new terminology that is just more stuff to know/remember and is just inventing new domain terminology.

Reply:
`WaitingToDie` is the original name.  That terminology in a lot of places in the battle code.  So I'd rather not rename it just here.  It would need to be renamed in all the places as a separate PR.